iptables/table.go

```diff
@@ -1014,14 +987,37 @@ func (t *Table) applyUpdates() error {
 		t.logCxt.WithField("iptablesInput", input).Debug("Writing to iptables")
 
 		var outputBuf, errBuf bytes.Buffer
-		cmd := t.newCmd(t.iptablesRestoreCmd, "--noflush", "--verbose")
+		args := []string{"--noflush", "--verbose"}
+		if features.RestoreSupportsLock {
+			// Versions of iptables-restore that support the xtables lock also make it mandatory.  Make sure
+			// that we configure it to retry and configure for a short retry interval (the default is not to retry).
+			lockTimeout := t.lockTimeout.Seconds()
+			if lockTimeout <= 0 {
+				// Before iptables-restore added lock support, we were able to disable the lock completely, which
+				// was indicated by a value <=0 (and was our default).  Newer versions of iptables-restore require the
+				// lock so we override the default and set it to 10s.
+				lockTimeout = 10
+			}
+			lockProbeMicros := t.lockProbeInterval.Nanoseconds() / 1000
+			timeoutStr := fmt.Sprintf("%.0f", lockTimeout)
+			intervalStr := fmt.Sprintf("%d", lockProbeMicros)
+			args = append(args,
+				"--wait", timeoutStr, // seconds
+				"--wait-interval", intervalStr, // microseconds
+			)
+			log.WithFields(log.Fields{
+				"timeoutSecs":         timeoutStr,
+				"probeIntervalMicros": intervalStr,
+			}).Debug("Using native iptables-restore xtables lock.")
+		}
+		cmd := t.newCmd(t.iptablesRestoreCmd, args...)
 		cmd.SetStdin(&inputBuf)
 		cmd.SetStdout(&outputBuf)
 		cmd.SetStderr(&errBuf)
 		countNumRestoreCalls.Inc()
-		t.writeLock.Lock()
+		t.calicoXtablesLock.Lock() // Will be a dummy lock if our xtables lock is disabled.
```

Comment:
I'd prefer "Will be a dummy lock if iptables-restore is acquiring the xtables lock itself."

Reply:
There's more than one reason for it to be disabled but probably worth expanding the comment to explain.